### ISSUE TITLE
onDetach not called on all components in tree

### ISSUE THREAD

**Issue by mpilone** [vaadin/flow#5493]
I found that when the UI is closed/cleaned up via `UIInternalls.setSession(null)`, not all components in the tree get an onDetach call. This results in my application leaking resources. I attached a simple test case where I would expect to see the following output when the logout button is clicked:

- Bug view detached
- MyLayout detached
- MyButton detached

but I only ever see:

- Bug view detached

Stepping through the code I think it may be related to ComponentUtil.findComponents. It seems like that method should be returning all the components in the tree for a given element but the logic:

```
if (maybeComponent.isPresent()) {
            Component component = maybeComponent.get();
            componentConsumer.accept(component);
            return;
        }
```

returns before recursing into the child elements so only the root element (the view in this test case) is returned and detached in setSession via `ui.getChildren().forEach(ComponentUtil::onComponentDetach);`.

Configuration:

- Push: long polling
- Flow: 1.4.2
- Vaadin: 13.0.2
- Chrome: 73.0.3683.103

Test case attached.
[Bug 5493 Test Case.zip](https://github.com/vaadin/flow/files/3081062/Bug.5493.Test.Case.zip)

**Comment by mpilone** [comment]
This may be related to https://github.com/vaadin/flow/issues/4197

**Comment by mehdi-vaadin** [comment]
This issue was fixed in Vaadin 13.0.13 (Flow 1.4.8) by #6112.
